lib/utils/config.ts

```diff
@@ -17,16 +17,17 @@ import Promise = require('bluebird');
 import ResinSdk = require('resin-sdk');
 import _ = require('lodash');
 import deviceConfig = require('resin-device-config');
+import * as semver from 'resin-semver';
 
 const resin = ResinSdk.fromSharedOptions();
 
 export function generateBaseConfig(
 	application: ResinSdk.Application,
-	options: { appUpdatePollInterval?: number },
+	options: { version?: string; appUpdatePollInterval?: number },
 ) {
 	options = _.mapValues(options, function(value, key) {
 		if (key === 'appUpdatePollInterval') {
-			return value! * 60 * 1000;
+			return (<typeof options.appUpdatePollInterval>value)! * 60 * 1000;
```

Comment:
This seems a bit over-complicated. How about `options[key]!`?

Reply:
Oooh, great idea, done.